### PR TITLE
Take care of empty string id for findById, #113.

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Add IT test for empty string.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/SimpleDocumentDbRepository.java [hunk 1]
```diff
@@ -135,6 +135,11 @@ private PartitionKey createKey(String partitionKeyValue) {
     @Override
     public Optional<T> findById(ID id) {
         Assert.notNull(id, "id must not be null");
+
+        if (id instanceof String && !StringUtils.hasText((String) id)) {
```
Comment (sophiaso) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/SimpleDocumentDbRepository.java]: I think in another PR, `id` can also be `Integer` and `int` type? Not sure which one is correct. Is there any conflict?

Also, `String` and `Integer` are `Object` types, but `int` is primitive type, so we allow `int` type?

Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/support/SimpleDocumentDbRepository.java]: Yeah, both int and Integer are allowed. Here only String may has empty concept. take care of this here.
